src/components/actions/extendApplicationRoute.ts

```diff
@@ -0,0 +1,67 @@
+import { combineRequestParams } from '../utilityFunctions'
+import { PermissionPolicyType, Trigger } from '../../generated/graphql'
+import { DateTime } from 'luxon'
+import DBConnect from '../databaseConnect'
+import { getPermissionNamesFromJWT } from '../data_display/helpers'
+
+export const routeExtendApplication = async (request: any, reply: any) => {
+  const { applicationId, eventCode, extensionTime, data } = combineRequestParams(request, 'camel')
+
+  // Check permissions first -- currently we just check user has ANY Reviewing
+  // permissions for current template.
+  // TO-DO: define specific permission names required for extending deadlines
+  const { permissionNames: userPermissions } = await getPermissionNamesFromJWT(request)
+
+  const templatePermissions = (
```

Comment:
technically permissionNames

Reply:
It's the permissions per template though, not just the names. So it's strictly "template_permission_names", but since we use "templatePermissions" in the front-end and in the JWT, this seems consistent I thought.